Meson static linking with "thin libraries" is not supported on illumos

**Describe the bug**
The feature introduced in https://mesonbuild.com/Release-notes-for-0-60-0.html#more-efficient-static-linking-of-uninstalled-libraries does not check if so-called "thin libraries" are supported on the system.
Therefore static linking may be broken at least in some configurations.

**To Reproduce**
Building gst-plugins-bad1 fails with

```
[657/820] Linking target tests/check/elements_h263parse
FAILED: tests/check/elements_h263parse
/usr/gcc/7/bin/gcc  -o tests/check/elements_h263parse
tests/check/elements_h263parse.p/elements_h263parse.c.o
-L/usr/lib/libjpeg8-turbo/lib/amd64 -I/usr/X11/include/mesa
-I/usr/X11/include -I/usr/include/libjpeg8-turbo -Wl,-z -Wl,ignore -z
defs -m64 -R/usr/lib/libjpeg8-turbo/lib/amd64 -m64 -O2
-Wl,-L/usr/lib/libjpeg8-turbo/lib/amd64
-Wl,-R/usr/lib/libjpeg8-turbo/lib/amd64 -m64
'-Wl,-rpath,$ORIGIN/../../gst-libs/gst/basecamerabinsrc:$ORIGIN/../../gst-libs/gst/interfaces:$ORIGIN/../../gst-libs/gst/uridownloader:$ORIGIN/../../gst-libs/gst/codecparsers'
-Wl,--start-group
gst-libs/gst/basecamerabinsrc/libgstbasecamerabinsrc-1.0.so.0.1805.0
gst-libs/gst/interfaces/libgstphotography-1.0.so.0.1805.0
gst-libs/gst/uridownloader/libgsturidownloader-1.0.so.0.1805.0
tests/check/libparser.a
gst-libs/gst/codecparsers/libgstcodecparsers-1.0.so.0.1805.0 -lm
/usr/lib/amd64/libgstreamer-1.0.so /usr/lib/amd64/libgobject-2.0.so
/usr/lib/amd64/libglib-2.0.so /usr/lib/amd64/libgstapp-1.0.so
/usr/lib/amd64/libgstbase-1.0.so /usr/lib/amd64/libgstpbutils-1.0.so
/usr/lib/amd64/libgstaudio-1.0.so /usr/lib/amd64/libgstvideo-1.0.so
/usr/lib/amd64/libgsttag-1.0.so /usr/lib/amd64/libgstcontroller-1.0.so
/usr/lib/amd64/libgstrtp-1.0.so /usr/lib/amd64/libgstcheck-1.0.so -lrt
/usr/lib/amd64/libgio-2.0.so -Wl,--end-group
ld: fatal: file tests/check/libparser.a: unknown file type
ld: fatal: file processing errors. No output written to
tests/check/elements_h263parse
collect2: error: ld returned 1 exit status
```

**Expected behavior**
No breakage :)

As a workaround I opened a PR against our userland component:

https://github.com/OpenIndiana/oi-userland/pull/7666

The patch disables "thin libraries" as for OSX:

https://github.com/OpenIndiana/oi-userland/blob/0d4dfb90167020df15aabd4e767b1f84b00dcdb2/components/developer/meson/patches/014-thin-libraries-breakage.patch

Thank you

**system parameters**
* Native build
* OpenIndiana x86_64 (illumos)
* Python version: 3.9.10
* meson version: 0.61.1
* ninja version: 1.10.2

--- a/mesonbuild/linkers/linkers.py
+++ b/mesonbuild/linkers/linkers.py
@@ -208,8 +208,12 @@ class ArLinker(ArLikeLinker):
         return self.can_rsp
 
     def get_std_link_args(self, env: 'Environment', is_thin: bool) -> T.List[str]:
-        # FIXME: osx ld rejects this: "file built for unknown-unsupported file format"
-        if is_thin and not env.machines[self.for_machine].is_darwin():
+        # Thin archives are a GNU extension not supported by the system linkers
+        # on Mac OS X, Solaris, or illumos, so don't build them on those OSes.
+        # OS X ld rejects with: "file built for unknown-unsupported file format"
+        # illumos/Solaris ld rejects with: "unknown file type"
+        if is_thin and not env.machines[self.for_machine].is_darwin() \
+          and not env.machines[self.for_machine].is_sunos():
             return self.std_thin_args
         else:
             return self.std_args

